### DISCUSSION TITLE
Can I debug standalone .Net Core tool with Visual Studio 2022?

### DISCUSSION THREAD

**Discussion by matteo-tosi** [security-code-scan/security-code-scan#260] [Q&A] [answered]
Greetings,
I would like to contribute to this tool as we use it in its standalone .Net Core version.
I am using Visual Studio 2022 and I am able to compile successfully.
When I launch the solution with the `SecurityCodeScan.Tool/.NET Core/security-scan` project as starter and execution go to `using (var workspace = MSBuildWorkspace.Create(properties))` I get this error:
![image](https://user-images.githubusercontent.com/39196215/188169049-3cbc3d4a-42a0-4428-8ffe-8b5c8e07bf29.png)
I think it depends on the fact that I use Visual Studio 2022. 
Ideas on how to fix it?

Thanks you,
Matteo

**Comment by JarLob** [comment, marked as answer]
I also use VS2022 and don't have this issue. Since you can reproduce I would appreciate if you found what is going on and fixed it. Maybe it is related to https://github.com/security-code-scan/security-code-scan/blob/039e760f860eec580d4bafc7e5e95dd1b4727d21/SecurityCodeScan.Tool/Program.cs#L268-L276 ?

**Reply by matteo-tosi** [reply]
Yes, the problem was caused by the fact that no instances are returned.
Specifically, I currently have `.NET Core 3.1` and `.NET 6.0` installed on my machine.
I did some tests with separate app console and if the app console points to `.NET 6.0` the istance is found without problems, if instead I use 3.1 or 5.0 the locator returns `null`.
The `SecurityCodeScan.Tool/.NET Core/security-scan` project references the following frameworks: `netcoreapp3.1; net5.0; net6.0`.
Order matters, so it takes the 3.1 framework and doesn't return instances. 
So much so that if I put only 6 everything works without problems.
At this point I don't know how you prefer to solve.
I hope I was helpful.

**Reply by JarLob** [reply]
Do you mean you choose which stand-alone exe to run from the output? I wonder how it works with the tool isntalled with `dotnet tool install --global security-scan`...

**Reply by JarLob** [reply]
I just pushed one thing. Try if it work for you.

**Reply by matteo-tosi** [reply]
Yes, it works. I have do this "fix" to run the project this afternoon, but
are you sure you would mantain version 3.1 and 5.0?
MSBuild.Locator don't return any visual studio instance, but i don't try
update this package.
Do you have only .Net 6.0 in your macchine?

**Reply by JarLob** [reply]
I have 3.1 and 6.0 (multiple minor versions) in sdk folder. Interestingly the locator returns only one - 3.1. It works, but fails to analyze. I think it is IDE only issue. By rearranging it the project settings I made it find 6.0.

**Reply by JarLob** [reply]
I intend to maintain 3.1 and 5.0 versions as long it doesn't bother me.
